cups/file.c

```diff
@@ -335,6 +335,132 @@ _cupsFileCheckFilter(
 #endif /* !_WIN32 */
 
 
+/*
+ * '_cupsFileGetConfAndComments()' - Get line and comments from a configuration file.
+ */
+
+char *					/* O  - Line read or @code NULL@ on end of file or error */
+_cupsFileGetConfAndComments(cups_file_t *fp,	/* I  - CUPS file */
+                char        *buf,	/* O  - String buffer */
+		size_t      buflen,	/* I  - Size of string buffer */
+                char        **value,	/* O  - Pointer to value */
+		int         *linenum)	/* IO - Current line number */
+{
+  char	*ptr;				/* Pointer into line */
+
+
+ /*
+  * Range check input...
+  */
+
+  DEBUG_printf(("2cupsFileGetConfAndComments(fp=%p, buf=%p, buflen=" CUPS_LLFMT
+                ", value=%p, linenum=%p)", (void *)fp, (void *)buf, CUPS_LLCAST buflen, (void *)value, (void *)linenum));
+
+  if (!fp || (fp->mode != 'r' && fp->mode != 's') ||
+      !buf || buflen < 2 || !value)
+  {
+    if (value)
+      *value = NULL;
+
+    return (NULL);
+  }
+
+ /*
+  * Read the next line...
+  */
+
+  *value = NULL;
+
+  while (cupsFileGets(fp, buf, buflen))
+  {
+    (*linenum) ++;
+
+    /*
+     * Remove the inline comment...
+     */
+     if ((ptr = strchr(buf, '#')) != NULL)
+     {
+       ptr--;
+       while(ptr >= buf)
+       {
+         // Null-terminate the string after the last non-whitespace
+         if(!_cups_isspace(*ptr))
+         {
+           *(ptr + 1) = '\0';
+           break;
+         }
+         ptr--;
+       }
+     }
+
+     /*
+      * Strip leading whitespace...
+      */
+
+      for (ptr = buf; _cups_isspace(*ptr); ptr ++);
+
+      if (ptr > buf)
+        _cups_strcpy(buf, ptr);
+
```

Comment:
The redundant part was else block, not whole if-else. Please put the if back.

Reply:
> 

Earlier I have used else block to print blank line as it is. If I just put if block than blank lines won't appears.